src/all-properties.ts

```diff
@@ -0,0 +1,64 @@
+import { Task } from '@ts-task/task';
+import { TaskValue, TaskError } from './type-helpers';
+
+export interface TaskObject {
+    [key: string]: Task<any, any>;
+}
+
+export type MapTaskValues <T extends TaskObject> = {
+    [P in keyof T]: TaskValue<T[P]>;
+};
+
+export type ObjectTaskErrors <T extends TaskObject, K extends keyof T> = TaskError<T[K]>;
+
+/**
+ * allProperties works similar to Task.all but for objects instead of array.
+ * It traverses an object whose values are Tasks and returns a new Task of the flattened object.
+ * The returned Task can fail with the sum of the individual errors.
+ * @param obj An object that has Task as values
+ */
+export function allProperties <T extends TaskObject, K extends keyof T>(obj: T): Task<MapTaskValues<T>, ObjectTaskErrors<T, K>> {
```

Comment:
I wonder if it couldn't be resolved based on `Task.all` 🤔. I guess it's possible using `Object.entries` (`Object.values` and `Object.keys`) as all of them sort their results in the same way (same order as doing a `for..in` loop on the object).